doc/specs/#4191 - Formatted Copy/spec.md

```diff
@@ -0,0 +1,90 @@
+
+### Settings option 1 - global setting
+
+We could have a global setting that adds the formatting when the user copies. When set, this would change the right click behavior.
+
+### Settings option 2 - key binding argument
+
+We could add an argument to the `copy` key binding argument to allow for formatted copying when the user chooses to do so.
+
+## UI/UX Design
+
+### Settings option 1 - global setting
+
+a. The user could list which kinds of formats they want included when they copy. When right clicking, they would copy with these formats. If this is set to something other than `plain`, holding `alt` and right clicking would copy plain text.
+
+`"copyFormats": ["html","rtf","plain"]`
+
+b. We could also just combine html and rtf into a single boolean. Users would either get plain text only (`false`) or all formatting (`true`) onto their clipboard. If this is set to `true`, the default right click behavior is reversed: right clicking copies the formatting and holding `alt` copies only the plain text.
+
+`"copyFormatting": true`
+
+### Settings option 2 - key binding argument
+
+a. Just like the `trimWhitespace` argument you can add to the `copy` key binding, we could add one for text formatting. This would not change the right click behavior.
+
+`{"command": {"action": "copy", "keepFormatting": true}, "keys": "ctrl+a"}`
+
+b. We could also split out the html and rtf formats. The right click behavior would still stay as default.
+
+`{"command": {"action": "copy", "formats": ["html","rtf","plain"]}, "keys": "ctrl+a"}`
+
+## Capabilities
+
+### Accessibility
+
+This shouldn't affect accessibility.
+
+### Security
+
+This does not affect security.
+
+### Reliability
+
+This does not affect reliability.
+
+### Compatibility
+
+This breaks the existing behavior of always copying the formatting. The justification for breaking this default behavior is in response to the community saying the default should be plain text only.
+
+### Performance, Power, and Efficiency
+
+## Potential Issues
+
+One possible issue is that discovering how to copy the formatting might be difficult to find. We could mitigate this by adding it into the settings.json file and commenting it out.
+
+## Future considerations
+
+We could always add an additional option if people want more flexibility. For example, if we ship a global setting now, we could ship a key binding later that lets you choose how you want to copy, and vice versa.
+
+## Conclusions
+
+The team has decided to have plain text as the default copy behavior and to enable formatted copying with a global setting that accepts a boolean value (settings option 1 - global setting, option b). In the future, we can modify this setting to also accept an array, so the user can specify which formats they would like to copy. Additionally, a key binding can be added to allow for greater flexibility.

```

Comment:
Wait didn't we decide on a combo of both 2 & 3?

We do the global setting _now_, as a boolean, because that's easy. (This is 2a)

We can then update that to _additionally_ accept an array in the future, where `true` == `[html, rtf, plain]` and `false`==`[plain]` (This is 2b)

Then, we could further add an optional parameter to the copy `ShortcutAction`. If it's set, we'll copy with the provided formatting. Otherwise, we'll use the value of the global setting? (This is 3)

Reply:
Okay I guess this is kinda what it _does_ say, I just didn't immediately parse that this was a phased implementation plan